tests/unit/moduleapi/internalsecret.tcl

```diff
@@ -0,0 +1,275 @@
+
+        # After unmarking the connection, internal commands fail.
+        r debug mark-internal-client unmark
+        assert_error {*unknown command*} {r internalauth.internalcommand}
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {Test `COMMAND *` commands with\without internal connections} {
+        # ------------------ Non-internal connection ------------------
+        # `COMMAND DOCS <cmd>` returns empty response.
+        assert_equal {} [r command docs internalauth.internalcommand]
+
+        # `COMMAND INFO <cmd>` should reply with null for the internal command
+        assert_equal {{}} [r command info internalauth.internalcommand]
+
+        # `COMMAND GETKEYS/GETKEYSANDFLAGS <cmd> <args>` returns an invalid command error
+        assert_error {*Invalid command specified*} {r command getkeys internalauth.internalcommand}
+        assert_error {*Invalid command specified*} {r command getkeysandflags internalauth.internalcommand}
+
+        # -------------------- Internal connection --------------------
+        # Non-empty response for non-internal connections.
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+
+        # `COMMAND DOCS <cmd>` returns a correct response.
+        assert_match {*internalauth.internalcommand*} [r command docs internalauth.internalcommand]
+
+        # `COMMAND INFO <cmd>` should reply with a full response for the internal command
+        assert_match {*internalauth.internalcommand*} [r command info internalauth.internalcommand]
+
+        # `COMMAND GETKEYS/GETKEYSANDFLAGS <cmd> <args>` returns a key error (not related to the internal connection)
+        assert_error {*ERR The command has no key arguments*} {r command getkeys internalauth.internalcommand}
+        assert_error {*ERR The command has no key arguments*} {r command getkeysandflags internalauth.internalcommand}
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {No authentication needed for internal connections} {
+        # Authenticate with a user that does not have permissions to any command
+        r acl setuser David on >123 &* ~* -@all +auth +internalauth.getinternalsecret
+        assert_equal {OK} [r auth David 123]
+
+        set reply [r internalauth.getinternalsecret]
+        assert_equal {OK} [r auth "internal connection" $reply]
+        # Execute a command for which David does not have permission
+        assert_equal {OK} [r internalauth.internalcommand]
+    }
+}
+
+start_cluster 1 0 [list config_lines $modules] {
+    set r [srv 0 client]
+
+    test {RM_Call of internal commands succeeds only for internal connections, and overrised user attachment} {
+        # Fail before authenticating as an internal connection.
+        assert_error {*unknown command*} {r internalauth.internal_rmcall internalauth.internalcommand}
```

Comment:
Also add a test and none internal command that runs RM_Call with internal command succeed (if used without the `C` flag).